public/src/components/teacher/CourseGroup.tsx

```diff
@@ -181,39 +148,37 @@ export class CourseGroup extends React.Component<ICourseGroupProps, ICourseGroup
         return <div>{names}</div>;
     }
 
-    private async handleUpdateStatus(gid: number, status: Group.GroupStatus): Promise<void> {
+    private async handleUpdateStatus(group: Group, status: Group.GroupStatus): Promise<void> {
         // delete or update group depending on the chosen status, then refresh the page
         status === Group.GroupStatus.DELETED ?
-            await this.deleteGroup(gid) :
-            await this.props.courseMan.updateGroupStatus(gid, status);
+            await this.deleteGroup(group) :
+            await this.props.courseMan.updateGroupStatus(group.getId(), status);
         this.props.navMan.refresh();
     }
 
-    private async deleteGroup(groupID: number) {
-        let withRepos = true;
-        let readyToDelete = true;
+    private async deleteGroup(group: Group) {
+        let readyToDelete = false;
```

Comment:
Could we make the logic clearer in this function by doing the following:
```typescript
let readyToDelete = group.getStatus() === Group.GroupStatus.PENDING
...
if (!readyToDelete && !isEmpty) {
  if (confirm(...))
  ...
```

Reply:
This way it will never delete a non-pending group with empty repo